### DISCUSSION TITLE
Packer Templates for macOS VMs

### DISCUSSION THREAD

**Discussion by notverypc** [macstadium/orka-desktop#2] [Q&A]
👋  I've been looking at Okra Desktop and wondered if there's any packer templates similar to what tart offers?
https://github.com/cirruslabs/macos-image-templates

**Comment by jasoncdavis0** [comment]
Thanks for the question @notverypc!

Today, we allow users to start either with 1) an IPSW (firmware file) from Apple, so it's like booting a brand new machine, or 2) using another OCI image that they have access too. Most people start with an IPSW file, login to the VM, and manually customize it to their needs. They can then use the "Push to OCI Registry" function to share the image with others.

We are planning to release a managed repository with a set of standard images and Packer templates for common OS versions, and are currently collecting use cases/requirements for this. 

Can you elaborate on your use case for Orka Desktop? 

3 common use cases we have heard so far that we are focused on are 1)MDM Profile testing, 2) Media Editing, 3) Build and test pipeline. Understanding your use case can help us prioritize the templates and images that are most important.

**Reply by notverypc** [reply]
Hey @jasoncdavis0 
I’m using VMs for MDM profile testing. 
My current workflow is to use a packer template that does all setup screens and creates a local user. 

I can pass different IPSW files to the template and get the same setup. 
This is an example https://github.com/notverypc/Tart_Templates/blob/main/macos_template.pkr.hcl

**Reply by jasoncdavis0** [reply]
Thanks for the feedback. MacStadium offers a packer plugin designed to work with our cluster product. We are evaluating approaches to bring support for Orka Desktop as well. I'll add a Feature Enhancement to our issues list to track, and we'll engage as we make progress.

**Comment by alexg-unity** [comment]
@jasoncdavis0 , We currently run Orka clusters in MacStadium EU/USA. Still, it would be good for our local devs to be able to use our CICD packer code  [Orka-Packer-Plugin](https://github.com/macstadium/packer-plugin-macstadium-orka) to build the customer master images on their local Orka desktop off the SAME base mac OSI's for quick test and development, rather than downloading 100's of GB from our image storage. thanks
